### PR TITLE
Add ethereum key generation

### PR DESCRIPTION
**Pull request by mvid**

### CODE REVIEW

**Review by zmanian — APPROVED** [review]

**Review by marbar3778 — APPROVED** [review]
LGTM

**Comment by marbar3778** [comment]
there seems to be some build issues. Can you fix this then we can merge this

File: module/cmd/peggy/cmd/eth_keys.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	"github.com/tendermint/tendermint/libs/cli"
+)
+
+const flagPassphrase = "passphrase"
+
+// Commands registers a sub-tree of commands to interact with
+// local private key storage.
+func Commands(defaultNodeHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "eth_keys",
```
Comment (jackzampolin) [inline module/cmd/peggy/cmd/eth_keys.go:25]: NIT: I really prefer typing `eth-keys` and the rest of the commands use `-`

File: module/cmd/peggy/cmd/eth_keys.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		Long: `Keyring management commands. Generated by the official Ethereum go library.
+
+The keyring supports the following backends:
+    test        Stores keys insecurely to disk. It does not prompt for a password to be unlocked
+                and it should be use only for testing purposes.
+`,
+	}
+
+	cmd.AddCommand(
+		AddKeyCommand(),
```
Comment (jackzampolin) [inline module/cmd/peggy/cmd/eth_keys.go:36]: This is fine with just `add` for now.

File: module/cmd/peggy/cmd/eth_keys.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+		}
+	}
+
+	return printCreate(cmd, keyOutput)
+}
+
+func printCreate(cmd *cobra.Command, keyOutput EthereumKeyOutput) error {
+	output, _ := cmd.Flags().GetString(cli.OutputFlag)
+
+	switch output {
```
Comment (jackzampolin) [inline module/cmd/peggy/cmd/eth_keys.go:113]: JSON and text output ++

File: module/cmd/peggy/cmd/eth_keys.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	case keys.OutputFormatText:
+		cmd.PrintErrln()
+		cmd.Println("private: %s public: %s address: %s", keyOutput.PrivateKey, keyOutput.PublicKey, keyOutput.Address)
```
Comment (jackzampolin) [inline module/cmd/peggy/cmd/eth_keys.go:116]:
```suggestion
		cmd.Printf("private: %s \npublic: %s \naddress: %s\n", keyOutput.PrivateKey, keyOutput.PublicKey, keyOutput.Address)
```

File: module/cmd/peggy/cmd/eth_keys.go (hunk 1 above)
Comment (mvid) [reply module/cmd/peggy/cmd/eth_keys.go:25]: I basically copied the cosmos `keys add` implementation. If we switch them, we should probably do all of them in peggy?

File: module/cmd/peggy/cmd/eth_keys.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	cmd.Flags().Bool(flags.FlagDryRun, false, "Perform action, but don't add key to local keystore")
+
+	cmd.SetOut(cmd.OutOrStdout())
+	cmd.SetErr(cmd.ErrOrStderr())
+
+	return cmd
+}
+
+type EthereumKeyOutput struct {
+	PublicKey  string `json:"private_key"`
```
Comment (jackzampolin) [inline module/cmd/peggy/cmd/eth_keys.go:67]:
```suggestion
	PublicKey  string `json:"public_key"`
```

File: module/cmd/peggy/cmd/eth_keys.go [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+	PrivateKey string `json:"public_key"`
```
Comment (jackzampolin) [inline module/cmd/peggy/cmd/eth_keys.go:68]:
```suggestion
	PrivateKey string `json:"private_key"`
```

File: module/cmd/peggy/cmd/eth_keys.go [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+	cmd.PersistentFlags().String(flags.FlagKeyringBackend, flags.DefaultKeyringBackend, "Select keyring's backend (os|file|test)")
+	cmd.PersistentFlags().String(cli.OutputFlag, "text", "Output format (text|json)")
+
+	return cmd
+}
+
+// AddKeyCommand defines a keys command to generate a key
+func AddKeyCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add",
```
Comment (jackzampolin) [inline module/cmd/peggy/cmd/eth_keys.go:50]: Maybe we change the name here to `generate` as we have no way of listing the generated keys? Adding full CRUD to this command tree isn't necessary right now.

Comment (mvid) [reply module/cmd/peggy/cmd/eth_keys.go:50]: Up to you, I was mostly trying to maintain symmetry with the `peggy keys add` command.

Comment (jackzampolin) [reply module/cmd/peggy/cmd/eth_keys.go:50]: This is fine we can merge as is

**Review by jackzampolin — APPROVED** [review]

**Review by jackzampolin — APPROVED** [review]

**Review by jkilpatr — APPROVED** [review]
@mvid  let me know when you think this is ready to merge 👍🏻

**Pull request merged**